tests/ui/job-view/fuzzy_test.jsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { mount } from 'enzyme/build';
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
+import configureMockStore from 'redux-mock-store';
+import { render, fireEvent, waitFor } from '@testing-library/react';
+
+import fuzzyJobList from '../mock/job_list/fuzzy_jobs/fuzzyJobList.json';
+import filteredFuzzyList from '../mock/job_list/fuzzy_jobs/filteredFuzzyList.json';
+import searchMochitest from '../mock/job_list/fuzzy_jobs/searchMochitest.json';
+import FuzzyJobFinder from '../../../ui/job-view/pushes/FuzzyJobFinder';
+
+const mockStore = configureMockStore([thunk]);
+
+describe('FuzzyJobFinder', () => {
+  const isOpen = true;
+  const decisionTaskId = 'YHKMjYZeSSmEZTrAPdRIag';
+  const id = 705037;
+  const currentRepo = {
+    id: 77,
+    repository_group: {
+      name: 'development',
+      description:
+        'Collection of repositories where code initially lands in the development process',
+    },
+    name: 'autoland',
+    dvcs_type: 'hg',
+    url: 'https://hg.mozilla.org/integration/autoland',
+    branch: null,
+    codebase: 'gecko',
+    description: 'The destination for automatically landed Firefox commits.',
+    active_status: 'active',
+    performance_alerts_enabled: true,
+    expire_performance_data: false,
+    is_try_repo: false,
+    tc_root_url: 'https://firefox-ci-tc.services.mozilla.com',
+    pushLogUrl: 'https://hg.mozilla.org/integration/autoland/pushloghtml',
+    revisionHrefPrefix: 'https://hg.mozilla.org/integration/autoland/rev/',
+  };
+  const store = mockStore({ fuzzyList: [] });
+  const testFuzzyJobFinder = mount(
+    <Provider store={store}>
```

Comment:
I'm curious why you're using the redux `Provider`? When I look at the `FuzzyJobFinder` component, I do not see that it is being passed props via `mapStateToProps`. The`FuzzyList` state is stored in that react component.

Don't use enzyme - [we have switched](https://bugzilla.mozilla.org/show_bug.cgi?id=1527237) to react-testing-library and haven't converted all of the old tests. I think you might be having issues because you are using mount and render (from the two different libraries). Just use react-testing-library and Jest. 

As you are writing tests, I suggest you start out as simply as possible. Get one line of code to work and then add another. Look at how some of the other tests that are only using react-testing-library are structured.

Reply:
@sarah-clements Thanks a lot for your advice
The things seem to be working well
But there is still an issue
I need to wait untill the function filterJobs has completely functioned so that 
and I am unable to currently think of any such algo so the fuze Search does its job fully and and I am  able to test things out

Reply:
I'm not sure what you're asking - can you be more specific?

I still see the redux Provider in your code. Per my first comment, I don't understand why you are using that.

Reply:
It might be easier to go over questions on a video call about writing tests with whomever is mentoring you on this task (not sure if that's @armenzg?)

Reply:
From the sound of things, it feels like the test updates the fuzzy modal's search field and triggers the filterJobs function by simulating an `enter` keypress, but then doesn't wait for filterJobs to actually filter the list of jobs before testing that it worked. filterJobs updates the lists stored in state and that state gets updated as the list of options in the UI, but that UI update can take some time to be reflected in the DOM, so the tests might be looking at a pre-update version of the list. 

I had suggested on Matrix that it might be sufficient to just wait X seconds (2 to 5 would probably be enough) between triggering filterJobs and expecting the list of jobs to be filtered.

Another less dirty option might be modifying filterJobs to emit some custom "job list filtered" event whenever it's sure to have a fully updated DOM version of the list, and then the test code could wait to observe that custom event before performing the actual tests.

Reply:
I tried removing the Provider but instead got this error
``` Invariant Violation: Could not find "store" in the context of "Connect(FuzzyJobFinder)". Either wrap the root component in a <Provider>, or pass a custom React context provider to <Provider> and the corresponding React context consumer to Connect(FuzzyJobFinder) in connect options.```
I have added one test of searching Linux
Please tell me if more have to be added
@KWierso , I have found another alternate solution for doing the same.
Thanks for guiding through it

Reply:
Can you pass an empty object to the store prop in the Provider? I think it's misleading to pass the `fuzzyList` here since state changes are not handled by redux in the `FuzzyJobFinder`.  If you search for `mapStateToProps` in your text editor you'll see how redux state is handled in different components (they're passed in as props) versus react state. :)